Sources/PCMultipartMessage.swift

```diff
@@ -0,0 +1,164 @@
+import Foundation
+import PusherPlatform
+
+public final class PCMultipartMessage {
+    public let id: Int
+    public let sender: PCUser
+    public let room:PCRoom
+    public let parts: [PCPart]
+    public let createdAt: String
+    public let updatedAt: String
+    
+    public var createdAtDate: Date { return PCDateFormatter.shared.formatString(self.createdAt) }
+    public var updatedAtDate: Date { return PCDateFormatter.shared.formatString(self.updatedAt) }
+    
+    public init(
+        id: Int,
+        sender: PCUser,
+        room: PCRoom,
+        parts: [PCPart],
+        createdAt: String,
+        updatedAt: String
+    ) {
+        self.id = id
+        self.sender = sender
+        self.room = room
+        self.parts = parts
+        self.createdAt = createdAt
+        self.updatedAt = updatedAt
+    }
+}
+
+extension PCMultipartMessage: CustomDebugStringConvertible {
+    public var debugDescription: String {
+        return "Message - ID: \(self.id), sender: \(self.sender.id)"
+    }
+}
+
+extension PCMultipartMessage: Hashable {
+    public var hashValue: Int {
+        return self.id
+    }
+    
+    public static func ==(lhs: PCMultipartMessage, rhs: PCMultipartMessage) -> Bool {
+        return lhs.id == rhs.id
+    }
+}
+
+public struct PCPart {
+    let payload: PCMultipartPayload
+}
+
+public struct PCMultipartInlinePayload {
+    let type: String
+    let content: String
+}
+
+public struct PCMultipartURLPayload {
+    let type: String
+    let url: String
+}
+
+public class PCMultipartAttachmentPayload {
+    let type: String
+    let size: Int
+    let name: String?
+    let customData: [String: Any]?
+    private let refresher: PCMultipartAttachmentURLRefresher
+    internal let refreshURL: String
+    internal var downloadURL: String
+    internal var expiration: String
+    
+     init(
+        type: String,
+        size: Int,
+        name: String?,
+        customData: [String: Any]?,
+        refresher: PCMultipartAttachmentURLRefresher,
+        refreshURL: String,
+        downloadURL: String,
+        expiration: String
+    ) {
+        self.type = type
+        self.size = size
+        self.name = name
+        self.customData = customData
+        self.refresher = refresher
+        self.refreshURL = refreshURL
+        self.downloadURL = downloadURL
+        self.expiration = expiration
+    }
+    
+    func url() -> String {
+        if (Date() > self.urlExpiry()) {
```

Comment:
I realise this is preparatory work (and I question it's place in this PR, but mike did this too so you're in good company 😉) but the spec suggests we should be refreshing the URL if it has less than 30 minutes of validity remaining.

> The url() method on AttachmentPayload must return a URL with at least 30 minutes of validity remaining.

Reply:
aaah! I was going along with Mike's android PR since this is the first bit of swift work I've done. I think I'm going to remove this so we can keep the PR limited to its scope. But will keep the promise comment in mind the next time I implement it.